x-pack/plugins/security_solution/server/endpoint/routes/resolver/queries/children.ts

```diff
@@ -59,6 +62,21 @@ export class ChildrenQuery extends ResolverQuery<ResolverEvent[]> {
 
   protected query(entityIDs: string[]): JsonObject {
     return {
+      /**
+       * Using collapse here will only return a single event per occurrence of a process.entity_id. The events are sorted
+       * based on timestamp in ascending order so it will be the first event that ocurred. The actual type of event that
+       * we receive for this query doesn't really matter (whether it is a start, info, or exec for a particular entity_id).
+       * All this is trying to accomplish is removing duplicate events that indicate a process existed for a node. We
+       * only need to know that a process existed and it's it's ancestry array and the process.entity_id fields because
+       * we will use it to query for the next set of descendants.
+       *
+       * The reason it is important to only receive 1 event per occurrence of a process.entity_id is it allows us to avoid
+       * ES 10k limit most of the time. If instead we received multiple events with the same process.entity_id that would
+       * reduce the maximum number of unique children processes we could retrieve in a single query.
+       */
+      collapse: {
```

Comment:
:information_source: nice

Reply:
:grey_question:  This won't do anything we don't expect like collapsing alerts or other things that maybe have the same process.entity_id right?

Reply:
Nope! This is only for children process nodes.